internal/policy/targets.go

```diff
@@ -96,6 +102,54 @@ func UpdateDelegation(targetsMetadata *tuf.TargetsMetadata, ruleName string, aut
 	return targetsMetadata, nil
 }
 
+// ReorderDelegations changes the order of delegations in TargetsMetadata.
+func ReorderDelegations(targetsMetadata *tuf.TargetsMetadata, ruleNames []string) (*tuf.TargetsMetadata, error) {
+	allDelegations := []tuf.Delegation{}
+	// Create a map of all delegations for quick validation.
+	rolesMap := make(map[string]tuf.Delegation)
+	for _, delegation := range targetsMetadata.Delegations.Roles {
+		rolesMap[delegation.Name] = delegation
+	}
+
+	ruleSet := set.NewSet[string]()
+
+	for _, ruleName := range ruleNames {
+		if ruleName == AllowRuleName {
```

Comment:
this brings up a good question: should users be specifying the allow-rule? IMO it's a convenience for us, so they should specify everything but.

Reply:
Indeed, I think the allow-rule is more like a default setting which sits below user-level abstraction. And `policy --list-rules` does not include the allow-rule either.